packages/devtools_app/lib/src/shared/editable_list.dart

```diff
@@ -5,24 +5,28 @@
 import 'package:flutter/foundation.dart';
 import 'package:flutter/material.dart';
 
+import '../primitives/utils.dart';
 import 'common_widgets.dart';
 import 'theme.dart';
 import 'utils.dart';
 
 class EditableList extends StatefulWidget {
-  const EditableList({
+  EditableList({
     required this.entries,
     required this.textFieldLabel,
     this.isRefreshing,
-    this.onEntryAdded,
-    this.onEntryRemoved,
+    Function(String)? onEntryAdded,
+    Function(String)? onEntryRemoved,
     this.onRefresh,
-  });
+  }) {
```

Comment:
can you do this in the constructor directly instead of the method body?

`)) : this.onEntryAdded = onEntryAdded ?? (entry) => entries.add(entry), ...`

Reply:
This took a while for Ben and I to work out properly, this was the only way we could find to do this while maintaining `final` on the callbacks.
The reason it is tricky is because we are accessing entries in the callback body

Reply:
kk so i gave it another whirl and it turns out that the right incantation of parens let me move it into the constructor initializer list :)